manticore/__main__.py

```diff
@@ -1,21 +1,44 @@
-
-import sys
-import logging
+"""
+This is the Manticore's CLI `manticore` script.
+"""
 import argparse
+import logging
+import sys
 
-from . import Manticore, STDIN_INPUT_DEFAULT_SIZE
-from .utils import log, config
-
-# XXX(yan): This would normally be __name__, but then logger output will be pre-
-# pended by 'm.__main__: ', which is not very pleasing. hard-coding to 'main'
-logger = logging.getLogger('manticore.main')
+from . import STDIN_INPUT_DEFAULT_SIZE
+from .utils import config, log
 
 consts = config.get_group('main')
 consts.add('recursionlimit', default=10000,
            description="Value to set for Python recursion limit")
 consts.add('timeout', default=0,
            description='Timeout, in seconds, for Manticore invocation')
 
+# XXX(yan): This would normally be __name__, but then logger output will be pre-
+# pended by 'm.__main__: ', which is not very pleasing. hard-coding to 'main'
+logger = logging.getLogger('manticore.main')
+
+
+def main():
+    """
+    Dispatches execution into one of Manticore's engines: evm or native.
+    """
+    args = parse_arguments()
+
+    if args.no_colors:
+        log.disable_colors()
+
+    sys.setrecursionlimit(consts.recursionlimit)
+
+    log.set_verbosity(args.v)
+
+    if args.argv[0].endswith('.sol'):
+        from manticore.ethereum.cli import main as evm_main
+        evm_main(args, logger)
+    else:
+        from manticore.native.cli import main as native_main
+        native_main(args, logger)
```

Comment:
IMHO, i'd rather import the above `main` functions without aliasing them; it might make grepping for them harder. I think `from a.b.c import main; main()` is pretty straightforward.

Reply:
The reason for that is that the `__main__.py` also contains `main` function, heh.

But I guess we can just name them as `native_main` and` evm_main`?